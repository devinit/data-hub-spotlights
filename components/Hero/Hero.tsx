import React, { FunctionComponent } from 'react';

interface HeroProps {
  title: string;
  excerpt?: string;
}

const Hero: FunctionComponent<HeroProps> = ({ excerpt, title }) => {
  return (
    <section className="hero hero--minor">
      <span className="hero__flourish"/>
      <span className="hero__flourish"/>
      <span className="hero__flourish"/>
      <span className="hero__flourish"/>
      <span className="hero__flourish"/>
      <span className="hero__flourish"/>
      <span className="hero__flourish"/>
      <span className="hero__flourish"/>
      <div className="row row--narrow">
        <div className="hero__content">
          <h1 className="hero__title"><span>{ title }</span></h1>
          { excerpt ? <p className="hero__excerpt"><span>{ excerpt }</span></p> : null }
        </div>
      </div>
    </section>
  );
};

export { Hero };
