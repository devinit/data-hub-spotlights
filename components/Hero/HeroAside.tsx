import React, { FunctionComponent } from 'react';
import classNames from 'classnames';

interface HeroAsideProps {
  className?: string;
}

const HeroAside: FunctionComponent<HeroAsideProps> = ({ children, className }) => {
  return <div className={classNames('hero__aside', className)}>{children}</div>;
};

export { HeroAside };
