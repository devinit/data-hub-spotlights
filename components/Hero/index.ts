export * from './Hero';
export * from './HeroAside';
