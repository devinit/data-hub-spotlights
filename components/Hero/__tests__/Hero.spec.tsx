/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { Hero } from '../Hero';

describe('Hero', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<Hero title="Spotlight" excerpt="My Spotlight" />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders without an excerpt when one is not provided', () => {
    const renderer = TestRenderer.create(<Hero title="Spotlight" />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
