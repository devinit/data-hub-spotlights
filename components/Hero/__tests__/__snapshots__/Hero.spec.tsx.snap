// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
<section
  className="hero hero--spotlight hero--minor"
>
  <span
    className="hero__flourish"
  />
  <span
    className="hero__flourish"
  />
  <span
    className="hero__flourish"
  />
  <span
    className="hero__flourish"
  />
  <span
    className="hero__flourish"
  />
  <span
    className="hero__flourish"
  />
  <span
    className="hero__flourish"
  />
  <span
    className="hero__flourish"
  />
  <div
    className="row row--narrow"
  >
    <div
      className="hero__content"
    >
      <h1
        className="hero__title"
      >
        <span>
          Spotlight
        </span>
      </h1>
      <p
        className="hero__excerpt"
      >
        <span>
          My Spotlight
        </span>
      </p>
    </div>
  </div>
</section>
`;

exports[`Hero renders without an excerpt when one is not provided 1`] = `
<section
  className="hero hero--spotlight hero--minor"
>
  <span
    className="hero__flourish"
  />
  <span
    className="hero__flourish"
  />
  <span
    className="hero__flourish"
  />
  <span
    className="hero__flourish"
  />
  <span
    className="hero__flourish"
  />
  <span
    className="hero__flourish"
  />
  <span
    className="hero__flourish"
  />
  <span
    className="hero__flourish"
  />
  <div
    className="row row--narrow"
  >
    <div
      className="hero__content"
    >
      <h1
        className="hero__title"
      >
        <span>
          Spotlight
        </span>
      </h1>
    </div>
  </div>
</section>
`;
